pollux/lib/vc-jwt/src/main/scala/io/iohk/atala/pollux/vc/jwt/JWTVerification.scala

```diff
@@ -24,25 +25,49 @@ object JWTVerification {
     "ES256" -> Set("ES256") // TODO: Only use valid type (added just for compatibility in the Demo code)
   )
 
-  def validateEncodedJwt[T](jwt: JWT, proofPurpose: Option[VerificationRelationship] = None)(
-      didResolver: DidResolver
-  )(decoder: String => Validation[String, T])(issuerDidExtractor: T => String): IO[String, Validation[String, Unit]] = {
-    val decodeJWT = Validation
-      .fromTry(JwtCirce.decodeRawAll(jwt.value, JwtOptions(false, false, false)))
-      .mapError(_.getMessage)
+  def validateAlgorithm(jwt: JWT): Validation[String, Unit] = {
+    val decodedJWT =
+      Validation
+        .fromTry(JwtCirce.decodeRawAll(jwt.value, JwtOptions(false, false, false)))
+        .mapError(_.getMessage)
+    for {
+      decodedJwtTask <- decodedJWT
+      (header, _, _) = decodedJwtTask
+      algorithm <- Validation
+        .fromOptionWith("An algorithm must be specified in the header")(JwtCirce.parseHeader(header).algorithm)
+      result <-
+        Validation
+          .fromPredicateWith("No PublicKey to validate against found")(
```

Comment:
as far as I know, public key might be missed in the header.
in this case, `kid` (key id) is used and should be resolved in the DID document.
@CryptoKnightIOG, could you check the JWT spec to clarify this case?